File: 2021/9/part1.rb
def calculate(file_name)
  lines = File.readlines(file_name).map(&:chomp)
  
  low_points = []

  lines.each_with_index do |line,y|
    line.chars.each_with_index do |point,x|
      point = point.to_i
      # puts "#{x},#{y} = #{point}"
      adjacent = []
      adjacent << lines[y][x-1].to_i if x-1 >= 0
      adjacent << lines[y-1][x].to_i if y-1 >= 0
      adjacent << lines[y][x+1].to_i if x+1 < line.length
      adjacent << lines[y+1][x].to_i if y+1 < lines.length
      low_points << point if adjacent.all? {|item| item > point}
    end
  end

  p low_points.reduce(0) {|a,b| a+b+1}
end

calculate("example_input.txt")
calculate("input.txt")

File: 2022/10/part1.py
"""Module"""
import functools


def calculate(file_name):
  """Method"""
  with open(file_name) as file:
    lines = [line.rstrip() for line in file]

  reg_X = 1
  cycle = 0
  signals = {}

  for line in lines:
    cycle += 1
    if (cycle-20) % 40 == 0:
      signals[cycle] = reg_X
    if line[0:4] == "addx":
      cycle += 1
      if (cycle-20) % 40 == 0:
        signals[cycle] = reg_X
      reg_X += int(line[5:])
  
  print(signals.items())
  answer = functools.reduce(lambda a,b: a+b[0]*b[1], signals.items(), 0)
  print(f"Answer: {answer}")


calculate("example_input.txt")
calculate("example_input_2.txt")
calculate("input.txt")
